### PR TITLE
util/set: add some useful utility functions for Set

### PR DESCRIPTION
**Pull request by noncombatant**
Also give each type of set its own file.

Updates #cleanup

### CODE REVIEW

**Comment by bradfitz** [comment]
Movement and changes in the same commit is hard to review.

File: util/set/set.go [hunk 1]
```diff
@@ -7,9 +7,38 @@ package set
 // Set is a set of T.
 type Set[T comparable] map[T]struct{}
 
+// New returns a new Set.
```
Comment (bradfitz) [inline util/set/set.go:10]: New generally returns a pointer in Go.

I'm not sure how much utility this constructor provides. The type is a map and we're not hiding that behind some opaque struct. Might as well reduce the number of was call sites can look.

Comment (noncombatant) [reply util/set/set.go:10]: Sounds good; removed.

File: util/set/set.go [hunk 2]
```diff
@@ -7,9 +7,33 @@ package set
 // Set is a set of T.
 type Set[T comparable] map[T]struct{}
 
+// NewFromSlice returns a new set constructed from the elements in slice.
+func NewFromSlice[T comparable](slice []T) Set[T] {
```
Comment (bradfitz) [inline util/set/set.go:11]: s/NewFromSlice/SetOf/ ? ala views.SliceOf or reflect.TypeOf/ValueOf? Then it avoids "New" again.

**Review by bradfitz — APPROVED** [review]

File: util/set/set.go (hunk 2 above)
Comment (noncombatant) [reply util/set/set.go:11]: Oh yeah, good idea. Coming up. Thanks!

**Pull request merged**